x-pack/solutions/observability/plugins/apm/public/components/app/metrics/static_dashboard/dashboards/dashboard_catalog.ts

```diff
@@ -5,52 +5,69 @@
  * 2.0.
  */
 
-export const AGENT_NAME_DASHBOARD_FILE_MAPPING: Record<string, string> = {
-  nodejs: 'nodejs',
-  'opentelemetry/nodejs': 'opentelemetry_nodejs',
-  'opentelemetry/nodejs/elastic': 'opentelemetry_nodejs',
-  java: 'java',
-  'opentelemetry/java': 'opentelemetry_java',
-  'opentelemetry/java/opentelemetry-java-instrumentation': 'opentelemetry_java',
-  'opentelemetry/java/elastic': 'opentelemetry_java',
-  'opentelemetry/dotnet': 'opentelemetry_dotnet',
-  'opentelemetry/dotnet/opentelemetry-dotnet-instrumentation': 'opentelemetry_dotnet',
-  'opentelemetry/dotnet/elastic': 'opentelemetry_dotnet',
-};
+// The new dashboard file names should be added here
+export const existingDashboardFileNames = new Set([
+  'classic_apm-apm-nodejs',
+  'classic_apm-apm-java',
+  'classic_apm-otel_other-nodejs',
+  'classic_apm-otel_other-java',
+  'classic_apm-otel_other-dotnet',
+  'otel_native-otel_other-nodejs',
+  'otel_native-otel_other-java',
+  'otel_native-otel_other-dotnet',
+]);
 
-/**
- * The specially formatted comment in the `import` expression causes the corresponding webpack chunk to be named. This aids us in debugging chunk size issues.
- * See https://webpack.js.org/api/module-methods/#magic-comments
- */
-export async function loadDashboardFile(filename: string): Promise<any> {
+// The new dashboard files should be mapped here
+// + changed with the new ones (following the naming convention)
+// + similar mapping for edot needed
+//     - example: otel_native-edot-nodejs
+export async function loadDashboardFile(filename: string) {
   switch (filename) {
-    case 'nodejs': {
+    case 'classic_apm-apm-nodejs': {
```

Comment:
These strings might be prone to errors. Maybe we could do something with types or constants? Just a suggestion.

Reply:
Those will match the name of the dashboard files when the new dashboards are ready so those cases will be: 
- classic_apm-apm-nodejs -> classic_apm-apm-nodejs.json
- otel_native-otel_other-java -> otel_native-otel_other-java.json 
And once we have the EDOT ones: 
- otel_native-edot-nodejs -> otel_native-edot-nodejs.json
I also created the Set with name (`existingDashboardFileNames`) so the filename will match those but I will check now how to improve that with stricter types/constants, good point!

Reply:
Note: To add that once the dashboard files are there and all the "names" are present